demo/checkpointing/train.py

```diff
@@ -0,0 +1,166 @@
+    parser.add_argument("--project",
+                        type=str,
+                        required=True,
+                        help="GCS project ID.")
+    parser.add_argument("--bucket",
+                        type=str,
+                        required=True,
+                        help="GCS bucket for saving checkpoints.")
+    parser.add_argument("--use-async",
+                        action="store_true",
+                        default=False,
+                        help="If checkpoint save should use async.")
+
+    # Hyper parameters
+    parser.add_argument("--embedding-dim",
+                        type=int,
+                        default=100,
+                        help="The size of each embedding vector.")
+    parser.add_argument("--hidden-dim",
+                        type=int,
+                        default=128,
+                        help="The number of features in the hidden state.")
+    parser.add_argument("--output-dim",
+                        type=int,
+                        default=35,
+                        help="The size of each output sample.")
+    parser.add_argument("--batch-size",
+                        type=int,
+                        default=32,
+                        help="How many samples per batch to load.")
+    parser.add_argument("--num-epochs",
+                        type=int,
+                        default=10,
+                        help="The number of full passes through training.")
+    parser.add_argument(
+        "--learning-rate",
+        type=float,
+        default=0.001,
+        help=
+        "The size of the optimizer steps taken during the training process.")
+    return parser.parse_args()
+
+
+def main(args: argparse.Namespace):
+    """
+    Typical usage example:
+
+      python3 -u demo/checkpointing/train.py \
+      --project=<gcs_project_id> \
+      --bucket=<bucket_name> \
+      --embedding-dim=100 \
+      --use-async
+    """
+
+    # Load dataset and create a Dataloader
+    dataset = WikiText2()
+    dataloader = DataLoader(dataset, batch_size=args.batch_size, num_workers=1)
+
+    # Create Model
+    model = TextClassifier(dataset.vocab_size, args.embedding_dim,
+                           args.hidden_dim, args.output_dim)
+    ckpt = CheckpointHelper(args.project, args.bucket)
+
+    # Define the loss function and optimizer
+    criterion = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.learning_rate)
+
+    losses = []
+    save_checkpoint_times = []
+    total_time = time.time()
+
+    # Training workload
+    for epoch in range(args.num_epochs):
+        model.train()
+        for inputs, labels in dataloader:
+            optimizer.zero_grad()
+            outputs = model(inputs)
+            labels = labels.type(torch.FloatTensor)
+            loss = criterion(outputs, labels)
+            loss.backward()
+            optimizer.step()
+
+        losses.append(loss.detach().numpy())
```

Comment:
Why is losses being tracked here? It doesn't seem like we use this data.

Reply:
Whoops! This was left over from previous work where I was collecting losses for printing visualization charts using matplotlib.